### PR TITLE
[UIE-161] dedupe library dependencies

### PR DESCRIPTION
**Pull request by cahrens**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/UIE-161

When @kpierre13 and I ran `yarn dedupe` on her recent library security update, we encountered this large number of changes and 3 files with build errors, mostly due to the spread operator.

To make future library upgrades easier, I thought it was best to go ahead and submit the dedupe separately. While fixing the typescript errors in 2 of the files, I converted them to TSX as it was trivial to do so. The third file I just added ignores to as the code is minimally used and likely to go away.

### CODE REVIEW

File: src/components/CloudProviderIcon.tsx [hunk 1]
```diff
@@ -7,15 +7,12 @@ type CloudProviderIconProps = {
   cloudProvider: CloudProvider;
 } & JSX.IntrinsicElements['svg'];
 
-export const CloudProviderIcon = ({ cloudProvider, ...props }: CloudProviderIconProps) => {
-  const icon = {
+export const CloudProviderIcon = (props: CloudProviderIconProps): ReactNode => {
```
Comment (cahrens) [inline src/components/CloudProviderIcon.tsx:10]: Example usage:

image

File: src/components/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Spinner, SpinnerProps } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { CSSProperties, ReactNode } from 'react';
+
+export { icon } from '@terra-ui-packages/components';
+
+export const centeredSpinner = (props: SpinnerProps = {}): ReactNode => {
```
Comment (cahrens) [inline src/components/icons.tsx:7]: This really should be capitalized, but I didn't want to change all the files that call this.

Example usage:

image

File: src/components/icons.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Spinner, SpinnerProps } from '@terra-ui-packages/components';
```
Comment (cahrens) [inline src/components/icons.tsx:1]: Git didn't recognize this as a rename due to the high percentage of lines that changed.

File: src/components/icons.tsx (hunk 2 above)
Comment (nawatts) [reply src/components/icons.tsx:7]: I would suggest keeping this as-is, since it's used as a render function and not as a component.

**Review by nawatts — APPROVED** [review]

**Review by kpierre13 — APPROVED** [review]

**Pull request merged**